### PR TITLE
MGMT-12187 Fix Logout

### PR DESCRIPTION
**Pull request by mareklibra**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please select the appropriate options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: ui/frontend/src/components/Page/HeaderTools.tsx [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
-    form.target = 'hidden-form';
-    const url = new URL(tokenEndpointResult.token_endpoint);
-    form.action = `${url.protocol}//${url.host}/logout`;
-    document.body.appendChild(form);
+      const input = document.createElement('input');
+      input.type = 'hidden';
+      input.name = 'then';
+      // The /logout oauth endpoint will redirect (HTTP 302) here
+      input.value = getAuthorizationEndpointUrl();
+      console.log('After succeful logout, the flow will continue here: ', input.value);
```
Comment (achuzhoy) [inline ui/frontend/src/components/Page/HeaderTools.tsx:36]: misspell:
successful

File: ui/frontend/src/components/Page/HeaderTools.tsx [hunk 2]
```diff
@@ -1,38 +1,56 @@
...
-    console.log('Logout: submitting POST request to logout');
-    form.submit();
+      form.appendChild(input);
+
+      const url = new URL(tokenEndpointResult.token_endpoint);
+      const actionUrl = `${url.protocol}//${url.host}/logout`;
+      form.action = actionUrl;
+      document.body.appendChild(form);
+
+      console.log('About to submitt POST: ', actionUrl);
```
Comment (achuzhoy) [inline ui/frontend/src/components/Page/HeaderTools.tsx:45]: misspell: submit

File: ui/backend/src/k8s/oauth.ts [hunk 3]
```diff
@@ -144,6 +145,13 @@ export async function logout(req: Request, res: Response): Promise<void> {
     logger.error(err);
   }
 
+  // try {
```
Comment (achuzhoy) [inline ui/backend/src/k8s/oauth.ts:148]: is this commented out block needed?

File: ui/deployProd.sh [hunk 4]
```diff
@@ -1,29 +1,31 @@
...
-echo ${APP_URL}
-
-./createTlsSecret.sh
-
-cat deployment.yaml |
-    sed "s/___NAMESPACE___/${NAMESPACE}/g" |
-    sed "s/___IMAGE___/${IMAGE}/g" |
-    sed "s/___ROUTE_HOST___/${ROUTE_HOST}/g" |
-    sed "s/___APP_URL___/${APP_URL}/g" |
+export EDGE_INGRESS_NAME=$(oc get ingresses.config.openshift.io cluster -o json| jq -j .spec.domain)
```
Comment (achuzhoy) [inline ui/deployProd.sh:7]: To get rid of the pipe to jq command, I'd use:
oc get ingresses.config.openshift.io cluster -o jsonpath={'.spec.domain'}

File: ui/backend/src/k8s/oauth.ts (hunk 3 above)
Comment (mareklibra) [reply ui/backend/src/k8s/oauth.ts:148]: Nope, removing.

File: ui/deployProd.sh (hunk 4 above)
Comment (mareklibra) [reply ui/deployProd.sh:7]: ok, thanks.

File: ui/frontend/src/components/Page/HeaderTools.tsx (hunk 2 above)
Comment (mareklibra) [reply ui/frontend/src/components/Page/HeaderTools.tsx:45]: Eagle eye :-)

File: ui/frontend/src/components/Page/HeaderTools.tsx (hunk 1 above)
Comment (mareklibra) [reply ui/frontend/src/components/Page/HeaderTools.tsx:36]: Done

**Review by alknopfler — APPROVED** [review]
LGTM

**Pull request merged**